compiler/arm/codegen/OMRTreeEvaluator.cpp

```diff
@@ -1454,20 +1454,64 @@ TR::Register *OMR::ARM::TreeEvaluator::conversionAnalyser(TR::Node          *nod
 
 static void signExtend(TR::Node *node, TR::Register *dst, TR::Register *src, int32_t bitsInDst, TR::CodeGenerator *cg)
    {
```

Comment:
`signExtend` and `zeroExtend` are largely the same function, save for three opcode differences (unless I'm missing something subtle).  In the interests of code size (always a concern on ARM), would it be better to have a single `generateSignOrZeroExtend` function that is parameterized by those opcodes and sharing the bulk of the logic?  The other code generators do similar things in other contexts.

Reply:
Updated the code following the review comment.